staging/src/k8s.io/apimachinery/pkg/runtime/serializer/protobuf/protobuf_test.go

```diff
@@ -66,3 +71,155 @@ func (t *mockTyper) ObjectKinds(obj runtime.Object) ([]schema.GroupVersionKind,
 func (t *mockTyper) Recognizes(_ schema.GroupVersionKind) bool {
 	return false
 }
+
+func TestSerializerEncodeWithAllocator(t *testing.T) {
+	testCases := []struct {
+		name    string
+		gvk     schema.GroupVersionKind
+		creater runtime.ObjectCreater
+		obj     runtime.Object
+	}{
+		// scenario 1
+		{
+			name:    "roundtrip a bufferedMarshaller obj",
+			gvk:     schema.GroupVersionKind{Group: "group", Version: "version", Kind: "Carp"},
+			creater: &mockCreater{obj: &testapigroupv1.Carp{}},
+			obj: &testapigroupv1.Carp{
+				TypeMeta: metav1.TypeMeta{APIVersion: "group/version", Kind: "Carp"},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "name",
+					Namespace: "namespace",
+				},
+				Spec: testapigroupv1.CarpSpec{
+					Subdomain: "carp.k8s.io",
+				},
+			},
+		},
+
+		// scenario 2
+		{
+			name:    "roundtrip a runtime.Unknown obj",
+			gvk:     schema.GroupVersionKind{Group: "group", Version: "version", Kind: "Unknown"},
+			creater: &mockCreater{obj: &runtime.Unknown{}},
+			obj:     &runtime.Unknown{TypeMeta: runtime.TypeMeta{APIVersion: "group/version", Kind: "Unknown"}},
```

Comment:
Let's also set some Raw data in it.

Reply:
it will make the test more complicated as we would have to compare the decoded object stored in the Raw data to something else that `obj`.  My intention was just to test various cases in the encoder. If we want to test the input perhaps we should add some fuzzer tests.

Reply:
This is related to my other comment - I don't care about decoding.

We should be comparing whether the serialized version of data is the same (which mitigates your concern above).

Reply:
it will work only if we remove the first Decoding call as well,  otherwise, it will fail.

Reply:
Yes - I guess I wasn't explicit enough, but I would actually remove decoding completely.